src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift

```diff
@@ -205,17 +205,20 @@ class ExposureSubmissionCoordinator: NSObject, ExposureSubmissionCoordinating, R
 				isLoading(true)
 				self.model.exposureSubmissionService.getTemporaryExposureKeys { error in
 					isLoading(false)
-
-					if let error = error {
-						// User selected "Don't Share" / "Nicht teilen"
-						if case .notAuthorized = error {
-							self.model.exposureSubmissionService.isSubmissionConsentGiven = false
-							self.showTestResultScreen(with: testResult)
-						} else {
-							self.showErrorAlert(for: error)
-						}
-					} else {
+					
+					guard let error = error else {
 						self.showTestResultScreen(with: testResult)
+						return
+					}
+					
+					self.model.exposureSubmissionService.isSubmissionConsentGiven = false
```

Comment:
Do we really want to retract the consent if the framework throws an error other than `.notAuthorized`?

Reply:
Very good point. No, only on `.notAuthorized`. Changed 👍